lib/improver/tests/utilities/test_TemporalInterpolation.py

```diff
@@ -219,7 +501,44 @@ def test_valid_interpolation_from_given_list(self):
             self.cube_time_0, self.cube_time_1)
         expected_data = np.ones((self.npoints, self.npoints)) * 4
         expected_time = self.time_extra.timestamp()
-        expected_fp = 3
+        expected_fp = 3 * 3600
+
+        self.assertArrayAlmostEqual(expected_data, result.data)
+        self.assertArrayAlmostEqual(
+            result.coord('time').points, expected_time, decimal=5)
+        self.assertAlmostEqual(result.coord('forecast_period').points[0],
+                               expected_fp)
```

Comment:
Could we add a datatype check `self.assertEqual(result.data.dtype, np.float32)` to this and the two tests below please.  (Potentially for time coords as well, but that can be part of the `check_coord_bounds` discussion.)

Reply:
Added.